### ISSUE TITLE
GitHub.ino code with JSON6 book - Http status: -1

### ISSUE THREAD

**Issue by fscii** [bblanchon/ArduinoJson#1532]
Sorry for noob question but I bought the book JSON 6 in spring 2020, not sure if that means its outdated since I'm only starting on it now.  That being said...

Using provided code example "GitHub.ino" on an ESP8266 d1 mini in Arduino IDE 1.8.13 and ArduinoJSON 6.17.3 also ESP8266 version 2.7.4

downloaded sketchbook6.zip files.  GitHub.ino says

// This program was tested with:
//   - Arduino 1.8.10
//   - ESP8266 code 2.6.3

I followed book instructions to get GitHub key (pp94-95) and also update certificate thumbprint (pp 97).
Result:  Unexpected HTTP status: -1

GitHub shows my token has never been used. I deleted it, made a new one, put that one in the .ino and same result.
https://github.com/settings/tokens
ArduinoJson tutorial — public_repo      NEVER USED

I have checked the FAQ and here to see if anyone had mentioned this and found nothing.

MUCH thanks in advance and apologies if I posted this in the wrong place (advise where it should be since I didn't see a discussion thread here).

Note: trying it in VsCode->PlatformIO gave a warning.
'bool HTTPClient::begin(String, String)' is deprecated (declared at C:\Users\fscii\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266HTTPClient\src/ESP8266HTTPClient.h:177) [-Wdeprecated-declarations]

**Comment by bblanchon** [comment]
Hi @fscii,

First, thanks a lot for buying my book, it means a lot to me!

Also, thank you for reporting this issue.
Indeed, I get the same error.

After enabling the log in HTTPClient, I can see:

```
[HTTP-Client][begin] url: https://api.github.com/graphql
[HTTP-Client][begin] host: api.github.com port: 443 url: /graphql
[HTTP-Client][begin] httpsFingerprint: 968407df0b1cf65814dfd7333557519b154d8ce7
[HTTP-Client][sendRequest] type: 'POST' redirCount: 0
[HTTP-Client] connected to api.github.com:443
[HTTP-Client] transport level verify failed
[HTTP-Client][returnError] error(-1): connection failed
``` 

I'm looking for a solution...

Best regards,
Benoit

**Comment by bblanchon** [comment]
Switching to the new API fixes the issue!

Instead of:

```c++
http.begin(F(GITHUB_URL), F(GITHUB_FINGERPRINT));
```

you must do:

```c++
WiFiClientSecure client;
client.setFingerprint(GITHUB_FINGERPRINT);
http.begin(client, F(GITHUB_URL));
```

This also removes the deprecation warning.

I'll upload a new `sketchbook6.zip` in a minute.

**Comment by bblanchon** [comment]
@fscii, did you try the new version?

**Comment by fscii** [comment]
@bblanchon yes I did and it works.  Thank you very much for helping.
I'm going through each example as I get time.  I did notice also SPIFFS deprecated and in the esp01 example using SPIFFS I simply did a "search replace" with LittleFS and it works perfectly.  Onto the Adafruit example, can get it all to work except that map.

Maybe its useful for a noob to go through all of this with noob eyes lol

Thanks again!
